scripts/pr_previews/utils.py

```diff
@@ -28,6 +29,21 @@ def configure_logging() -> None:
     )
 
 
+def setup_git_account() -> None:
+    run_subprocess(["git", "config", "user.name", "github-actions[bot]"])
+    run_subprocess(
+        ["git", "config", "user.email", "github-actions[bot]@users.noreply.github.com"]
+    )
+
+
+@contextmanager
+def switch_branch(branchname: str) -> None:
+    run_subprocess(["git", "fetch", "origin", branchname])
+    run_subprocess(["git", "switch", branchname])
+    yield
+    run_subprocess(["git", "checkout", "-"])
```

Comment:
This fails in GitHub actions. You could still keep this function but have the logging after the yield

https://github.com/Qiskit/documentation/actions/runs/9900538685/job/27351493566#step:7:17

Reply:
Are you sure? It looks like the job succeeds: https://github.com/Qiskit/documentation/actions/runs/9909483488/job/27377633039

It did fail when I used `switch` though.